### PR TITLE
Service Professions Overhaul

### PR DESCRIPTION
**Pull request by Crimsonerva**
## Description of changes
The service sector of the game is currently very lacking. Sure, you can repeatedly grow and grind crops for biomass, but the diversity is low on this end. By implementing gameplay loops revolving around individualistic steps of cooking, raising livestock, making bartending actually interesting, etc, hopefully we can alleviate a lot of these issues.

This PR also introduces the concept of tailoring. Clothing quality should degrade over time, but that's not something I can feasibly do in my coding skills yet. For now, I've taken all clothing out of lathes to implement this new tailoring mechanic. In order to make clothing now, you must now print "frames" or "stencils" of different clothings. In order to complete these new stencils, a variety of gear is available from a new lathe: the tailoring bench. Much like an autolathe... except fabricates tailoring items. After applying tools to a stencil, you'll then be able to apply a corresponding material to complete the stencil and get the clothing you've just designed. In the future, I'd like things to have to get tailor fit, or at least have different sizes according to players, but once again, not feasible with my level of coding skills.

With the new cooking, instead of mashing everything inside of a microwave, you'll now be hand crafting ingredients for the most part. When it comes to adding toppings (like for pizza) or specialized flavors (cake), you'll include them in the microwave, depending on what it is. Here, hopefully using sprites, we can make it intuitive (and also via descriptions) on the next steps, and step away from requiring the codex in this sector. While the codex is a powerful tool, I want to make sure everything I can in the service sector is intuitive and can be either learned from trial and error, or through clues left behind in the game.

Please make sure to look the locations of icons for their licensing agreements.

## Authorship
me, me, and all me!! (mostly)
new cooking code in cookers (now appliances) are from aurora, along with the bulk of the new food sprites
my time (hours) spent fixing all the errors
aside from that - me.

**Hydroponics:**
- [x] Reorganize biogenerator
- [ ] Add custom names to the list (so bottles wont just show up as "bottle"
- [ ] Un-grayscale hydroponics (they cannot currently export; as the items are all created from datums)
- [ ] Re-organize existing plants and sprites
- [ ] Add aquaponics

**Bartending:**
- [ ] Readjust all mixed drinks to be able to be made from scratch
- [ ] Make recipes a bit more unique, perhaps delayed, instead of instantly transforming
- [ ] Give mixed drinks export value on starter planet, and at least 2 beacons.

**Cooking:**
- [x] Resprite all current food (thats not LRP, and remove LRP)
- [x] Add new tools for food prep work (hamburger press, cookie cutter, mixing bowl, mixer, etc)
- [x] Move all food (most) prep to hand-work, instead of using the microwave.
- [x] Convert the microwave into a dual stove-oven so you don't have to keep baking cakes in the microwave
- [x] Remove the cookers (candy machine, oven, etc) from the game and use new tools for prep work and creation
- [x] Ensure all food has an actual crafting recipe
- [x] Add MORE food

**Ranching:**
- [ ] Create new meals requiring butchering of livestock
- [x] Change chicken limit (since they need constant maintenance anyways or will die)
- [x] Make all livestock available via biogenerator (until we get lifecycles there's no reason to make them purchased by beacons)
- [ ] Make at least 1 beacon have some livestock exports.
- [ ] Resprite livestock
- [ ] Add baby goats and cows
- [x] Make biogenerator provide apiarist necessities.

**Tailoring:**
- [ ] add tailoring bench
- [ ] make stencils for all types of clothing (that isnt armor, at least, advanced)
- [ ] make stencils for clothing attachments
- [ ] make tools for tailoring bench
- [ ] make tools applicable on stencils
- [ ] make stencils actually turn into clothing
- [ ] make each type of livestock produce a different textile, used in various clothings.

### CODE REVIEW

**Comment by Crimsonerva** [comment]
fyi i accidentally removed area control boards in a commit and i couldnt readd them without it making another commit and i had no way to delete it, so. dont yell at me about it please!!

**Comment by Crimsonerva** [comment]
an issue i noticed that seems unrelated is that corn is actually really evil and grows off of the tray, then sprouts tendrils and starts attacking you. i havent seen any reason it does this but just marking it down as an FYI

**Comment by Crimsonerva** [comment]
a lot of this is old code which i found way easier to port; will have to port newer code to get up-to-date in the future

File: code/_macros.dm [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 #define PUBLIC_GAME_MODE SSticker.master_mode
 
-#define CLAMP01(x) 		(clamp(x, 0, 1))
+#define Clamp(value, low, high) (value <= low ? low : (value >= high ? high : value))
+
+#define CLAMP01(x) 		(Clamp(x, 0, 1))
```
Comment (out-of-phaze) [inline code/_macros.dm]: NO
```suggestion
#define CLAMP01(x) 		(clamp(x, 0, 1))
```

File: code/modules/fabrication/machines/_cooker.dm [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+		QUEUE_TEMPERATURE_ATOM(src) // cool every tick if we're not turned on or heating
+	if(use_power != POWER_USE_OFF)
+		heat_up()
+	. = ..()
+
+/obj/machinery/appliance/cooker/update_cooking_power()
+	var/temp_scale = 0
+	if(temperature > min_temp)
+		if(temperature >= optimal_temp)
+			temp_scale = Clamp(1 - ((optimal_temp - temperature) / optimal_temp), 0, 1)
```
Comment (out-of-phaze) [inline code/modules/fabrication/machines/_cooker.dm]:
```suggestion
			temp_scale = clamp(1 - ((optimal_temp - temperature) / optimal_temp), 0, 1)
```

File: code/modules/fabrication/machines/_cooker_output.dm [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	return name
+
+/obj/item/chems/food/variable/get_name_sans_prefix()
+	return jointext(splittext(..(), " ") - prefix, " ")
+
+/obj/item/chems/food/variable/proc/update_scale()
+	if (reagents?.total_volume)
+		var/ratio = reagents.total_volume / size
+		scale = sqrt(ratio) //Scaling factor is square root of desired area
+		scale = Clamp(scale, min_scale, max_scale)
```
Comment (out-of-phaze) [inline code/modules/fabrication/machines/_cooker_output.dm]:
```suggestion
		scale = clamp(scale, min_scale, max_scale)
```

File: code/modules/fabrication/machines/stove.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	)
+
+/obj/machinery/appliance/cooker/stove/on_update_icon()
+	. = ..()
+	var/list/pans = list()
+	for(var/obj/item/chems/cooking_container/CC in contents)
+		var/image/pan_overlay
+		switch(CC.appliancetype)
+			if(APPLIANCE_SKILLET)
+				pan_overlay = image('icons/obj/cooking_machines.dmi', "skillet[Clamp(length(pans)+1, 1, 4)]")
```
Comment (out-of-phaze) [inline code/modules/fabrication/machines/stove.dm]:
```suggestion
				pan_overlay = image('icons/obj/cooking_machines.dmi', "skillet[clamp(length(pans)+1, 1, 4)]")
```

File: code/modules/fabrication/machines/stove.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+			if(APPLIANCE_SAUCEPAN)
+				pan_overlay = image('icons/obj/cooking_machines.dmi', "pan[Clamp(length(pans)+1, 1, 4)]")
```
Comment (out-of-phaze) [inline code/modules/fabrication/machines/stove.dm]:
```suggestion
				pan_overlay = image('icons/obj/cooking_machines.dmi', "pan[clamp(length(pans)+1, 1, 4)]")
```

File: code/modules/fabrication/machines/stove.dm [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+			if(APPLIANCE_POT)
+				pan_overlay = image('icons/obj/cooking_machines.dmi', "pot[Clamp(length(pans)+1, 1, 4)]")
```
Comment (out-of-phaze) [inline code/modules/fabrication/machines/stove.dm]:
```suggestion
				pan_overlay = image('icons/obj/cooking_machines.dmi', "pot[clamp(length(pans)+1, 1, 4)]")
```

File: code/_macros.dm [hunk 7]
```diff
@@ -2,7 +2,9 @@
 
 #define PUBLIC_GAME_MODE SSticker.master_mode
 
-#define CLAMP01(x) 		(clamp(x, 0, 1))
+#define Clamp(value, low, high) (value <= low ? low : (value >= high ? high : value))
+
```
Comment (out-of-phaze) [inline code/_macros.dm]: NO NO NO NO NO
https://www.byond.com/docs/ref/index.html#/proc/clamp
```suggestion
```

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
NO NO NO NO NO
https://www.byond.com/docs/ref/index.html#/proc/clamp

**Comment by Crimsonerva** [comment]
work atm. patched all the missing icons i could find and doesn't seem to break anything aside from part of the codex, which i can fix tonight

**Pull request merged**